Source/JavaScriptCore/heap/MarkedSpace.h

```diff
@@ -207,7 +207,7 @@ class MarkedSpace {
 
     Vector<Subspace*> m_subspaces;
 
-    std::unique_ptr<UncheckedKeyHashSet<HeapCell*>> m_preciseAllocationSet;
+    std::optional<UncheckedKeyHashSet<HeapCell*>> m_preciseAllocationSet;
```

Comment:
Is there a reason this is an optional at all? You can't have a JSGlobalObject without allocating at least a few precise allocations so this should effectively always be initialized.

Reply:
This set isn't created lazily when the first precise allocation is done. It's created if and when `enablePreciseAllocationTracking` is called. I don't know what the rationale is for having both modes.